src/toolbox/AreaOfInterestDrawer.js

```diff
@@ -252,38 +262,133 @@ class NgmAreaOfInterestDrawer extends I18nMixin(LitElement) {
+      attributes[i18next.t('Perimeter')] = `${props.perimeter}km`;
+      attributes[i18next.t('numberOfSegments')] = props.numberOfSegments;
+    } else if (props.type === 'line') {
+      attributes[i18next.t('Length')] = `${props.perimeter}km`;
+    }
+    return attributes;
+  }
+
+  getIconClass(id) {
+    const entity = this.interestAreasDataSource.entities.getById(id);
+    const type = entity.properties.type ? entity.properties.type.getValue() : undefined;
+    switch (type) {
+      case 'polygon':
+        return 'draw polygon icon';
+      case 'rectangle':
+        return 'vector square icon';
+      case 'line':
+        return 'route icon';
+      case 'point':
+        return 'map marker alternate icon';
+      default:
+        return '';
+    }
+  }
+
+
+  /**
+   * Adds AOI entity to data source
+   * @param attributes:
+   * {
+       id: string,
+       name: (optional) string,
+       show: boolean,
+       positions: Array<Cartesian3>,
+       area: (optional) string | number,
+       perimeter: (optional) string | number,
+       sidesLength: (optional) Array<string | number>,
+       numberOfSegments: (optional) number,
+       type: string<point | line | rectangle | polygon>
+   * }
+   */
+  addAreaEntity(attributes) {
+    const type = attributes.type;
+    const name = type.charAt(0).toUpperCase() + type.slice(1);
+    const entityAttrs = {
+      name: attributes.name || `${name} ${this.areasCounter_[type]}`,
+      show: typeof attributes.show === 'boolean' ? attributes.show : true,
+      properties: {
+        area: attributes.area,
+        perimeter: attributes.perimeter,
+        numberOfSegments: attributes.numberOfSegments,
+        sidesLength: attributes.sidesLength,
+        type: type,
+      }
+    };
+    if (type === 'rectangle' || type === 'polygon') {
+      entityAttrs.polygon = {
+        hierarchy: attributes.positions,
+        material: DEFAULT_AOI_COLOR
+      };
+    } else if (type === 'line') {
+      entityAttrs.polyline = {
+        positions: attributes.positions,
+        clampToGround: true,
+        width: 4,
+        material: DEFAULT_AOI_COLOR
+      };
+    } else if (type === 'point') {
+      entityAttrs.position = attributes.positions[0];
+      entityAttrs.point = {
+        color: DEFAULT_AOI_COLOR,
+        pixelSize: 6,
+        heightReference: HeightReference.CLAMP_TO_GROUND
+      };
+    }
+    return this.interestAreasDataSource.entities.add(entityAttrs);
+  }
+
+  showSectionModal(imageUrl) {
+    this.sectionImageUrl = imageUrl;
+    this.requestUpdate();
+  }
+
+  showAreaInfo(areaAttrs) {
+    const objectInfo = document.querySelector('ngm-object-information');
+    objectInfo.info = this.getInfoProps(areaAttrs);
+    objectInfo.opened = !!areaAttrs;
+    this.pickArea_(areaAttrs.id);
+  }
+
+  onAreaClick(event) {
+    if (event.target && event.target.type === 'checkbox') {
+      event.cancelBubble = true;
+    }
+  }
+
+  get getDrawState() {
```

Comment:
maybe a personal taste but this attribute should be renamed to `drawState` (`getXXX` are for functions)